compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -1045,6 +1047,78 @@ pub fn prohibit_assoc_item_binding(
             None
         },
     });
+
+    // Emit a suggestion if possible
+    if let Some((def_id, segment, _)) = segment
+        && segment.args().parenthesized == hir::GenericArgsParentheses::No
+        && let hir::TypeBindingKind::Equality { term: binding_arg_ty } = binding.kind
+    {
+        // Suggests removal of the offending equality constraint
+        let suggest_removal = |e: &mut Diag<'_>| {
+            let mut suggestion_span = binding.span.with_hi(binding.span.hi() + BytePos(1)); // Include the comma or the angle bracket at the end
+            if segment.args().bindings.len() == 1 {
+                // If it is the only binding specified
+                // include the starting angle bracket as well
+                suggestion_span = suggestion_span.with_lo(suggestion_span.lo() - BytePos(1))
+            }
+            if let Ok(suggestion) = tcx.sess.source_map().span_to_snippet(suggestion_span) {
+                e.span_suggestion_verbose(
+                    suggestion_span,
+                    "try removing this type binding",
```

Comment:
I will change the word "try" in the suggestion to "consider", but not "type binding" to "constraint" yet because if I do this is what the diagnostic will look like:

```
error[E0229]: associated type bindings are not allowed here
  --> tests/ui/associated-types/associated-types-eq-2.rs:85:10
   |
85 | impl Tr5<A = u32> for Bar {
   |          ^^^^^^^ associated type not allowed here
   |
   = note: `Tr5` is not a generic type
help: try removing this constraint
...
```

wherein we're calling the thing "type binding" in the error message but calling it "constraint" in the suggestion which will confuse the user.

Maybe we can make this change when you've renamed `TypeBinding' to `AssocItemConstraint`.

Reply:
Changed "try" to "consider"